codice/ddf
# DDF-4133 fix backup command

#### What does this PR do?
1. Fix the backup command so that it works with Solr basic auth
2. Consolidate the code to create an HTTP client for Solr so that it can be re-used more easily 

#### Who is reviewing it? 
@troymohl 
@tbatie 
@austinsteffes 

#### Select relevant component teams: 
@codice/solr 

-->
#### Ask 2 committers to review/merge the PR and tag them here.
@tbatie 
@vinamartin

#### How should this be tested?
1. Build it 
2. Install it
3. Run the `backup` command from Karaf console
4. Verify there are backup files in the directory `solr/server/solr/catalog/data`

#### Checklist:
- [NA] Documentation Updated
- [NA] Update / Add Threat Dragon models
- [NA] Update / Add Unit Tests
- [NA] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
